### PR TITLE
Fold SIMD-typed partial access to locals; delete base types from vector constants

### LINKED ISSUE #70052: Investigate removing gtSimdBaseJitType from GenTreeVecCon
https://github.com/dotnet/runtime/pull/68874 introduced a new `GenTreeVecCon` node and it currently carries a `gtSimdBaseJitType` field so that a handle can be reconstituted where required.

As per https://github.com/dotnet/runtime/pull/68874#discussion_r885010053, https://github.com/dotnet/runtime/pull/69971 likely removed the need to carry this information around and so we should remove the field if that is indeed the case.

### PR DESCRIPTION
**Pull request by SingleAccretion**
Enabling the folding of `SIMD`-typed indirs in local morph, it was observed that some CSEs were getting lost. The reason was that we always used `FLOAT`-derived handles for `TYP_SIMD` `IND`/`LCL_FLD` nodes, which would not be available in methods that only used `Vector128<int>` (for example). This change enhances the "canonical handle" logic to capture handles of other types to use for handle-less nodes.

We have positive diffs due to more CSEs, some light regressions due to more CSEs, and a couple large regressions due to more unrolling (`IND<simd>` appears cheaper to `gtSetEvalOrder` than `OBJ<simd>`).

Closes #70052.

### CODE REVIEW

**Comment by SingleAccretion** [comment]
@dotnet/jit-contrib

File: src/coreclr/jit/simd.cpp [hunk 1]
```diff
@@ -910,6 +910,30 @@ CorInfoType Compiler::getBaseJitTypeAndSizeOfSIMDType(CORINFO_CLASS_HANDLE typeH
...
+                pCanonicalHnd = &m_simdHandleCache->CanonicalSimd32Handle;
+                break;
+            default:
+                unreached();
+        }
+
+        if ((pCanonicalHnd != nullptr) && (*pCanonicalHnd == NO_CLASS_HANDLE))
+        {
+            *pCanonicalHnd = typeHnd;
+        }
```
Comment (jakobbotsch) [inline src/coreclr/jit/simd.cpp:936]: So essentially the first SIMD handle we see in this function becomes the canonical handle for this compilation?

**Review by jakobbotsch — APPROVED** [review]

File: src/coreclr/jit/simd.cpp (hunk 1 above)
Comment (SingleAccretion) [reply src/coreclr/jit/simd.cpp:936]: Yep. It's a workaround of course, but doing this "properly" would imply breaking the assumption that (most / all that are looked at) `varTypeIsStruct` locals have handles. Which is work that will be done at some point, but not now.

**Comment by jakobbotsch** [comment]
Failures are #76041 according to build analysis.

**Pull request merged**